controllers/sessionController.js

```diff
@@ -0,0 +1,49 @@
+const User = require("../models/User");
+const parseVErr = require("../utils/parseValidationErrs");
+
+const registerShow = (req, res) => {
+  res.render("register");
+};
+
+const registerDo = async (req, res, next) => {
+  if (req.body.password != req.body.password1) {
+    req.flash("error", "The passwords entered do not match.");
+    return res.render("register", {  errors: flash("errors") });
+  }
+  try {
+    await User.create(req.body);
+  } catch (e) {
+    if (e.constructor.name === "ValidationError") {
+      parseVErr(e, req);
+    } else if (e.name === "MongoServerError" && e.code === 11000) {
+      req.flash("error", "That email address is already registered.");
+    } else {
+      return next(e);
+    }
+    return res.render("register", {  errors: flash("errors") });
+  }
+  res.redirect("/");
+};
+
+const logoff = (req, res) => {
+  req.session.destroy(function (err) {
+    if (err) {
+      console.log(err);
+    }
+    res.redirect("/");
+  });
+};
+
+const logonShow = (req, res) => {
+  if (req.user) {
```

Comment:
Nice usage of an early return here! Alternatively, since it’s a single instruction for both the if/else conditions, you could also use the ternary operator:
```js
return req.user ? res.redirect("/") : res.render("logon");
```